cmd/gitops/install/cmd.go

```diff
@@ -122,9 +122,11 @@ func installRunCmd(cmd *cobra.Command, args []string) error {
 		providerClient := internal.NewGitProviderClient(osysClient.Stdout(), osysClient.LookupEnv, auth.NewAuthCLIHandler, log)
 
 		gitClient, gitProvider, err = factory.GetGitClients(context.Background(), providerClient, services.GitConfigParams{
-			URL:       installParams.ConfigRepo,
-			Namespace: namespace,
-			DryRun:    installParams.DryRun,
+			// We need to set URL and ConfigRepo to the same value so a deploy key is created for public config repos
+			URL:        installParams.ConfigRepo,
+			ConfigRepo: installParams.ConfigRepo,
```

Comment:
I dont love having to set these two values but this is the only solution short of rewriting the entire GetGitClients functionality and all the places that use it.